### PR TITLE
Keep LibreSSL 3.4.0 up-to-date for OpenBSD -current

### PR DESCRIPTION
**Pull request by gonzalo-**
I hope this is enough:

```
test result: ok. 46 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 8.95s

   Doc-tests openssl-sys

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s
```

Cheers.-

### CODE REVIEW

**Comment by sfackler** [comment]
Looks like you need to run cargo fmt.

**Comment by sfackler** [comment]
It looks like LibreSSL 3.4.0 hasn't released yet, so this will need to wait until it does.

**Comment by gonzalo-** [comment]
> It looks like LibreSSL 3.4.0 hasn't released yet, so this will need to wait until it does.

sure, no problem.

**Comment by vishwin** [comment]
3.4.0 is now released.

All of the guards that were for 3.3.2 and later need to stay that way instead of being redefined for 3.4.0. Otherwise the 3.3 branch, which is the stable branch, will most likely break.

File: openssl-sys/build/cfgs.rs [hunk 1]
```diff
@@ -34,6 +34,12 @@ pub fn get(openssl_version: Option<u64>, libressl_version: Option<u64>) -> Vec<&
         if libressl_version >= 0x3_03_02_00_0 {
             cfgs.push("libressl332");
         }
+        if libressl_version >= 0x3_03_03_00_0 {
```
Comment (vishwin) [inline openssl-sys/build/cfgs.rs:37]: This is not necessary unless an actual guard condition in some other area of code is needed.

File: openssl-sys/build/main.rs [hunk 2]
```diff
@@ -233,6 +233,8 @@ See rust-openssl README for more information:
             (3, 3, 0) => ('3', '3', '0'),
             (3, 3, 1) => ('3', '3', '1'),
             (3, 3, _) => ('3', '3', 'x'),
+            (3, 4, 0) => ('3', '4', '0'),
+            (3, 4, _) => ('3', '4', 'x'),
```
Comment (vishwin) [inline openssl-sys/build/main.rs:237]: Do not add 3.4.x until the 3.4 branch is considered stable.

File: openssl-sys/src/ssl.rs [hunk 3]
```diff
@@ -315,7 +315,7 @@ cfg_if! {
     if #[cfg(ossl102)] {
         pub const SSL_OP_NO_DTLSv1: c_ulong = 0x04000000;
         pub const SSL_OP_NO_DTLSv1_2: c_ulong = 0x08000000;
-    } else if #[cfg(libressl332)] {
+    } else if #[cfg(libressl340)] {
```
Comment (vishwin) [inline openssl-sys/src/ssl.rs:318]: This guard needs to stay at 3.3.2.

File: openssl/src/ssl/mod.rs [hunk 4]
```diff
@@ -211,14 +211,14 @@ bitflags! {
 
         /// Disables the use of DTLSv1.0
         ///
-        /// Requires OpenSSL 1.0.2 or LibreSSL 3.3.2 or newer.
-        #[cfg(any(ossl102, ossl110, libressl332))]
+        /// Requires OpenSSL 1.0.2 or LibreSSL 3.4.0 or newer.
+        #[cfg(any(ossl102, ossl110, libressl340))]
```
Comment (vishwin) [inline openssl/src/ssl/mod.rs:215]: This guard needs to stay at 3.3.2.

File: openssl/src/ssl/mod.rs [hunk 5]
```diff
@@ -211,14 +211,14 @@ bitflags! {
...
         const NO_DTLSV1 = ffi::SSL_OP_NO_DTLSv1;
 
         /// Disables the use of DTLSv1.2.
         ///
-        /// Requires OpenSSL 1.0.2 or LibreSSL 3.3.2 or newer.
-        #[cfg(any(ossl102, ossl110, libressl332))]
+        /// Requires OpenSSL 1.0.2 or LibreSSL 3.4.0 or newer.
+        #[cfg(any(ossl102, ossl110, libressl340))]
```
Comment (vishwin) [inline openssl/src/ssl/mod.rs:221]: This guard needs to stay at 3.3.2.

File: openssl-sys/build/main.rs [hunk 6]
```diff
@@ -273,7 +275,7 @@ fn version_error() -> ! {
         "
 
 This crate is only compatible with OpenSSL 1.0.1 through 1.1.1, or LibreSSL 2.5
-through 3.3.x, but a different version of OpenSSL was found. The build is now aborting
+through 3.4.x, but a different version of OpenSSL was found. The build is now aborting
```
Comment (vishwin) [inline openssl-sys/build/main.rs:278]: Should be 3.4.0 for now, per above comment.

**Review by vishwin — CHANGES_REQUESTED** [review]

**Comment by hargoniX** [comment]
I'm currently trying to get some program that depends on openssl to work on -current and it's failing at the moment since this PR isn't merged yet. Is there any chance you'll get to addressing the review in the next days? @gonzalo- otherwise I'd happily take over the PR and fix it for you.

**Comment by gonzalo-** [comment]
@hargoniX please take over, i will be off a few days, sorry guys

**Comment by hargoniX** [comment]
Continued in #1524

**Pull request closed without merging**